src/cmd/sv-rollout/svrestarter.go

```diff
@@ -58,26 +61,28 @@ func (s *SvRestarter) Restart() error {
 	case <-restartDone:
 		if err != nil {
 			if strings.Contains(string(out), "timeout: run: ") {
-				err = ErrRestartTimeout{Service: s.Service}
+				rerr = ErrRestartTimeout{Service: s.Service}
 				tags = append(tags, "status:timeout")
 			} else {
-				err = ErrRestartFailed{Service: s.Service, Message: string(out)}
+				rerr = ErrRestartFailed{Service: s.Service, Message: string(out)}
 				tags = append(tags, "status:success")
 			}
 		}
 	case <-s.preempt:
 		<-preemptionAcceptable
-		err = ErrRestartPreempted{Service: s.Service}
+		rerr = ErrRestartPreempted{Service: s.Service}
 		tags = append(tags, "status:preempted")
 	}
 
-	if Statsd != nil {
-		tags = append(tags, "service:"+s.Service)
-		Statsd.Timer("service.restart", time.Since(start), tags, 1)
-	}
+	go func() {
```

Comment:
> Interesting, I thought it would be cheap compared to the actual library call and all that but dunnolol on that.

Well the call is happening anyway no? just on another thread plus the potential alloc of the goroutine stack ( I _think_  )

> making the above race condition worse

I don't think we should depend on timing to alleviate races